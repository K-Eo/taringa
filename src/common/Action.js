import React from 'react'
import classNames from 'classnames'

const Action = props => (
  <button
    {...props}
    className={classNames('btn btn-action', props.className)}
  >
    { props.icon && <i className={`fa fa-${props.icon}`}/> }
    {props.children}
  </button>
)

export default Action
