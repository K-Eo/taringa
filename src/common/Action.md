Basic Action

```js
<Action>
  click me
</Action>
```

Action with icon

```js
<Action icon="comments">
  click me
</Action>
```
