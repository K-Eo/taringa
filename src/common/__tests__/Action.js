import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Action from '../Action'

describe('Action', () => {
  it('renders default', () => {
    const tree = renderer.create(<Action>click me</Action>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with icon', () => {
    const tree = renderer
      .create(<Action icon="comments">click me</Action>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('handles onClick', () => {
    const mock = jest.fn()

    const action = shallow(<Action onClick={mock} />)
    action.find('button').simulate('click')
    expect(mock).toBeCalled()
  })
})
