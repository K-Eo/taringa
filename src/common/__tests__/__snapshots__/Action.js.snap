// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action does not render if children is zero 1`] = `
<button
  className="btn btn-light btn-sm foo"
  id="foo-action"
>
  <i
    className="fa fa-close"
  />
</button>
`;

exports[`Action passes additional props 1`] = `
<button
  className="btn btn-light btn-sm foo"
  id="foo-action"
>
  <i
    className="fa fa-close"
  />
  <span
    className="ml-2"
  >
    Foobar
  </span>
</button>
`;

exports[`Action renders 1`] = `
<button
  className="btn btn-light btn-sm"
/>
`;

exports[`Action renders content 1`] = `
<button
  className="btn btn-light btn-sm"
>
  <span
    className=""
  >
    Foobar
  </span>
</button>
`;

exports[`Action renders content with icon 1`] = `
<button
  className="btn btn-light btn-sm"
>
  <i
    className="fa fa-close"
  />
  <span
    className="ml-2"
  >
    Foobar
  </span>
</button>
`;

exports[`Action renders with additional class 1`] = `
<button
  className="btn btn-light btn-sm foo"
>
  <i
    className="fa fa-close"
  />
  <span
    className="ml-2"
  >
    Foobar
  </span>
</button>
`;
